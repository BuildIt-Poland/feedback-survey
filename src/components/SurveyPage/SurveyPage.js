import React, { useState, useEffect, useContext } from 'react';

import { surveyMatchParams } from '../../types';
import Page from '../Layout/Page';
import Main from '../App/Main';
import { SurveyContext } from '../../context/SurveyContext';
import Loading from '../Loading';
import ErrorQuestionsMessage from '../Survey/ErrorQuestionsMessage';
import { NO_QUESTIONS } from '../../constatnts/messages';
import SurveyFormWithFormik from '../SurveyFormWithFormik';

const SurveyPage = ({ match = { params: {} } }) => {
  const { data, isLoading, error } = useContext(SurveyContext);
  const [surveyId, setSurveyID] = useState('');
  const [employeeName, setEmployeeName] = useState('');
  const [surveyData, setSurveyData] = useState(data);

  useEffect(() => {
    setSurveyID(match.params.surveyId ? match.params.surveyId : '');
    setEmployeeName(match.params.employeeName ? match.params.employeeName : '');
    setSurveyData(data);
  });

  if (isLoading) {
    return <Loading />;
  }

  if (error) {
    return <ErrorQuestionsMessage message={error} />;
  }

  if (surveyData && !surveyData.questions.length) {
    return <ErrorQuestionsMessage message={NO_QUESTIONS} />;
  }

  return (
    <Main>
      <Page>
        <SurveyFormWithFormik surveyData={surveyData} surveyId={surveyId} employeeName={employeeName} />
      </Page>
    </Main>
  );
};

SurveyPage.propTypes = {
  match: surveyMatchParams
};

export default SurveyPage;
