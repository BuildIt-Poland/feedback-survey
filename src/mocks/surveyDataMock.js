export const surveyDataMock = {
  surveyData: {
    questions: [
      {
        id: '4',
        type: '',
        content: 'Test question?',
        required: false,
        name: 'testName',
        answerType: 'kindOfFeedback'
      }
    ],
    answerTypes: [
      {
        type: 'kindOfFeedback',
        values: ['monthly', 'quaterly', 'annual']
      }
    ]
  },
  errors: {},
  values: {}
};
