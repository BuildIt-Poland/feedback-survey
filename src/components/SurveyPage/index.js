import { withRouter } from 'react-router-dom';

import SurveyPage from './SurveyPage';

export default withRouter(SurveyPage);
