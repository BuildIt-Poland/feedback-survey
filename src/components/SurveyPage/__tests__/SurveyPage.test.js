import React from 'react';
import { create } from 'react-test-renderer';

import SurveyPage from '../SurveyPage';
import { SurveyContext } from '../../../context/SurveyContext';
import { surveyDataMock } from '../../../mocks/surveyDataMock';

const contextValue = {
  isLoading: false,
  error: '',
  data: surveyDataMock.surveyData
};

describe('COMPONENT - SurveyPage', () => {
  it('renders correctly with params', () => {
    const match = {
      params: {
        surveyId: '12345',
        employeeName: 'John'
      }
    };
    const component = create(
      <SurveyContext.Provider value={contextValue}>
        <SurveyPage match={match} />
      </SurveyContext.Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('renders Loading component if data loading', () => {
    const props = {
      ...contextValue,
      isLoading: true
    };
    const component = create(
      <SurveyContext.Provider value={props}>
        <SurveyPage />
      </SurveyContext.Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders error message', () => {
    const props = {
      ...contextValue,
      error: 'error message'
    };
    const component = create(
      <SurveyContext.Provider value={props}>
        <SurveyPage />
      </SurveyContext.Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders info message if there are no questions ', () => {
    const props = {
      ...contextValue,
      data: {
        questions: [],
        answerTypes: []
      }
    };
    const component = create(
      <SurveyContext.Provider value={props}>
        <SurveyPage />
      </SurveyContext.Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with data', () => {
    const component = create(
      <SurveyContext.Provider value={contextValue}>
        <SurveyPage />
      </SurveyContext.Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
