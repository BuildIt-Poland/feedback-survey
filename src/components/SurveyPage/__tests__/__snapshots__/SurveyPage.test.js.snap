// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - SurveyPage renders Loading component if data loading 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 hUosMD"
>
  <h3
    className="sc-htpNat gwZFiw StyledHeading-sc-1rdh4aw-0 gRvHvF"
  >
    Loading...
  </h3>
</div>
`;

exports[`COMPONENT - SurveyPage renders correctly with data 1`] = `
<main
  className="sc-bwzfXH kUHKbp"
>
  <section
    className="sc-bdVaJa jihNFi"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="StyledGrommet-sc-19lkkz7-0 eUfLDp"
      >
        <div
          className="sc-dnqmqq eGMfmX"
        >
          <div
            className="sc-htoDjs fULGFU FormField__FormFieldBox-m9hood-0 igkoAa StyledBox-sc-13pk1d4-0 CAsMy"
          >
            <label
              className="StyledText-sc-1sadyjn-0 bMXxti"
            >
              Test question?
            </label>
            <div
              className="StyledBox-sc-13pk1d4-0 gZSidW"
            >
              <div
                className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 yIdXy"
              >
                <input
                  autoComplete="off"
                  className="StyledTextInput-sc-1x30a0s-0 cyWwXE"
                  name="testName"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="StyledBox-sc-13pk1d4-0 dNgQiZ"
      >
        <button
          className="sc-bxivhb ecUKhc StyledButton-sc-323bzc-0 bmExQD"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          size="large"
          type="submit"
        >
          submit
        </button>
      </div>
    </form>
  </section>
</main>
`;

exports[`COMPONENT - SurveyPage renders correctly with params 1`] = `
<main
  className="sc-bwzfXH kUHKbp"
>
  <section
    className="sc-bdVaJa jihNFi"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="StyledGrommet-sc-19lkkz7-0 eUfLDp"
      >
        <div
          className="sc-dnqmqq eGMfmX"
        >
          <div
            className="sc-htoDjs fULGFU FormField__FormFieldBox-m9hood-0 igkoAa StyledBox-sc-13pk1d4-0 CAsMy"
          >
            <label
              className="StyledText-sc-1sadyjn-0 bMXxti"
            >
              Test question?
            </label>
            <div
              className="StyledBox-sc-13pk1d4-0 gZSidW"
            >
              <div
                className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 yIdXy"
              >
                <input
                  autoComplete="off"
                  className="StyledTextInput-sc-1x30a0s-0 cyWwXE"
                  name="testName"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="StyledBox-sc-13pk1d4-0 dNgQiZ"
      >
        <button
          className="sc-bxivhb ecUKhc StyledButton-sc-323bzc-0 bmExQD"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          size="large"
          type="submit"
        >
          submit
        </button>
      </div>
    </form>
  </section>
</main>
`;

exports[`COMPONENT - SurveyPage renders error message 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 ePVEiU"
>
  <h4
    className="StyledHeading-sc-1rdh4aw-0 btqNMh"
  >
    error message
  </h4>
</div>
`;

exports[`COMPONENT - SurveyPage renders info message if there are no questions  1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 ePVEiU"
>
  <h4
    className="StyledHeading-sc-1rdh4aw-0 btqNMh"
  >
    Oops, something went wrong, there is no questions.
  </h4>
</div>
`;
